src/client/js/AppShell.jsx

```diff
@@ -0,0 +1,33 @@
+import { px, rgb } from 'csx'
+import React from 'react'
+import { Router as ServerRouter, Switch } from 'react-router'
+import { BrowserRouter } from 'react-router-dom'
+import { style } from 'typestyle'
+import { routes } from '../routes'
+import { Navigation } from './components/Navigation'
+import { RouteWithData } from './components/RouteWithData'
+import { debugClassName } from './styles/common'
+
+const appShellClassName = style(debugClassName('app'), {
```

Comment:
I though the whole point of `typestyle` was to be used it with `TypeScript`, which we are clearly not using. Given that, I think the whole library is just adding unnecessary overhead and bloating the bundle. Doing CSS with tagged template strings could do the same job without any of that eg. https://github.com/stackcss/sheetify (browserify plugin, there must be a webpack equivalent out there)

Reply:
No, not only.
Typestyle will generate also in plain JS the needed CSS classes. And even without JS editors (like code) can statically analyze the properties.